nlp_primitives/number_of_words_in_quotes.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+from string import punctuation
+
+import pandas as pd
+from featuretools.primitives import TransformPrimitive
+from nltk import word_tokenize
+from woodwork.column_schema import ColumnSchema
+from woodwork.logical_types import IntegerNullable, NaturalLanguage
+
+
+class NumberOfWordsInQuotes(TransformPrimitive):
+    """Determines the number of words in quotes in a string.
+
+    Description:
+        Given a list of strings, determine the number of words in quotes
+        in each string.
+
+        This implementation handles Unicode characters.
+
+        If a string is missing, return `NaN`.
+
+    Args:
+        capture (str, optional): Specify what kind of quotes to match.
+        Argument "single" matches on only single quotes (' ').
+        Argument "double" matches words between double quotes (" ").
+        Argument "both" matches words between either kind of quotes.
+        Defaults to "both".
+
+    Examples:
+         >>> x = ['"python" java prolog', '"this is a string"', '"binary" "ternary"']
+        >>> number_of_words_in_quotes = NumberOfWordsInQuotes()
+        >>> number_of_words_in_quotes(x).tolist()
+        [1, 4, 2]
+    """
+
+    name = "number_of_words_in_quotes"
+    input_types = [ColumnSchema(logical_type=NaturalLanguage)]
+    return_type = ColumnSchema(logical_type=IntegerNullable, semantic_tags={"numeric"})
+    default_value = 0
+
+    def __init__(self, capture="both"):
+        if capture not in ["both", "single", "double"]:
+            raise ValueError(
+                f"{capture} is not a valid argument. Specify 'both', 'single', or 'double'"
+            )
+        self.capture = capture
+
+    def get_function(self):
+        def _word_tokenize(text):
+            tokens = word_tokenize(text)
+            ct = 0
+            for word in tokens:
+                if len(word.strip(punctuation).strip()) > 0:
+                    ct += 1
+            return ct
```

Comment:
I agree that the comprehension is neater, but I'm concerned about the performance implications of allocating that much continuous memory. On `NumberOfUniqueWords`, I had a 20% perf improvement on large strings by replacing a comprehension with iteration and a `ct` variable.

Reply:
That's fair.

Reply:
What about just `return sum(len(word.strip(punctuation + " ")) > 0 for word in word_tokenize(text))`? I think that might use a generator internally instead of allocating memory for a list. Not sure how that performance would compare...or if that makes it less readable.

Reply:
Was just thinking about a similar thing! Will try it out now